applications/crossbar/src/crossbar_view.erl

```diff
@@ -0,0 +1,1165 @@
+        {RespJObj, 'json'} when is_list(RespJObj) ->
+            {Req2, Context3} = chunk_send_jsons({Req, Context2}, RespJObj),
+            {length(RespJObj), LoadMap#{started_chunk => cb_context:fetch(Context3, 'started_chunk')
+                                       ,context => Context3
+                                       ,cowboy_req => Req2
+                                       ,queried_jobjs => QueriedJObjs ++ RespJObj
+                                       }
+            };
+        {SentLength, 'csv'} when is_integer(SentLength) ->
+            {SentLength, LoadMap#{started_chunk => 'true'
+                                 ,context => Context2
+                                 ,cowboy_req => Req1
+                                 }
+            };
+        {Resp, 'csv'} when is_list(Resp), StartedChunk ->
+            'ok' = cowboy_req:chunk(Resp, Req),
+            {length(Resp), LoadMap#{context => Context2}};
+        {Resp, 'csv'} when is_list(Resp), not StartedChunk ->
+            {Req2, Context3} = init_chunk_stream({Req, Context2}, 'csv'),
+            'ok' = cowboy_req:chunk(Resp, Req2),
+            {length(Resp), LoadMap#{started_chunk => 'true'
+                                   ,context => Context3
+                                   ,cowboy_req => Req2
+                                   }
+            }
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Apply filter function if provided while keep maintaining the result
+%% order.
+%% Filter function can be arity 1 (operating on list of JObjs) and
+%% arity 2 (classical Erlang foldl function, e.g. (JObj, Acc)).
+%%
+%% For arity 1, the passed JObjs is in the reverse order, reverse
+%% you're response at the end.
+%% @end
+%%--------------------------------------------------------------------
+-spec apply_filter(mapper_fun(), kz_json:objects()) -> kz_json:objects().
+apply_filter('undefined', JObjs) ->
+    lists:reverse(JObjs);
+apply_filter(Mapper, JObjs) when is_function(Mapper, 1) ->
+    %% Can I trust you to return sorted result in the correct direction?
+    Mapper(JObjs);
+apply_filter(Mapper, JObjs) when is_function(Mapper, 2) ->
+    [JObj
+     || JObj <- lists:foldl(Mapper, [], JObjs),
+        not kz_term:is_empty(JObj)
+    ].
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Apply user specified mapper.
+%% See send_chunked_mapper/3 comment
+%%
+%% Note: Keep the original passed JOBjs ORDER in your mapper!
+%% @end
+%%--------------------------------------------------------------------
+-spec apply_chunked_mapper(chunked_mapper_fun(), load_params(), kz_json:objects(), ne_binary()) ->
+                                  chunked_mapper_ret().
+apply_chunked_mapper('undefined', #{context := Context
+                                   ,chunk_response_type := 'json'
+                                   }, JObjs, _) ->
+    {JObjs, Context};
```

Comment:
This clause would crash line 795 above I think.

Reply:
I think you either need to update `chunked_mapper_ret()` to include the cowboy_req record or remove the pattern match at line 795

Reply:
I forgot to change `chunked_mapper_ret()`. fixed now.